content/sensu-go/5.17/api/secrets.md

```diff
@@ -0,0 +1,386 @@
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "env"
+  },
+  "spec": {
+  }
+}' \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/providers/env
+
+HTTP/1.1 200 OK
+{{< /highlight >}}
+
+#### API Specification {#providers-provider-put-specification}
+
+/providers/:provider (PUT) | 
+----------------|------
+description     | Creates or updates the specified provider. The provider resource and API version cannot be altered.
+example URL     | http://hostname:8080/api/enterprise/secrets/v1/providers/env
+payload         | {{< highlight shell >}}
+{
+  "type": "Env",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "env"
+  },
+  "spec": {
+  }
+}
+{{< /highlight >}}
+response codes  | <ul><li>**Success**: 201 (Created)</li><li>**Malformed**: 400 (Bad Request)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+
+### `/providers/:provider` (DELETE) {#providers-provider-delete}
+
+The `/providers/:provider` API endpoint provides HTTP DELETE access to delete the specified provider from Sensu.
+
+#### EXAMPLE {#providers-provider-delete-example}
+
+The following example shows a request to the `/providers/:provider` API endpoint to delete the provider `env`, resulting in a successful HTTP `204 No Content` response.
+
+{{< highlight shell >}}
+curl -X DELETE \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN" \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/providers/env
+
+HTTP/1.1 204 No Content
+{{< /highlight >}}
+
+#### API Specification {#providers-provider-delete-specification}
+
+/providers/:provider (DELETE) | 
+--------------------------|------
+description               | Deletes the specified provider from Sensu.
+example url               | http://hostname:8080/api/enterprise/secrets/v1/providers/env
+response codes            | <ul><li>**Success**: 204 (No Content)</li><li>**Missing**: 404 (Not Found)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+
+## The `/secrets` endpoint
+
+### `/secrets` (GET)
+
+The `/secrets` API endpoint provides HTTP GET access to a list of secrets.
+
+#### EXAMPLE {#secrets-get-example}
+
+The following example demonstrates a request to the `/secrets` API endpoint, resulting in a list of secrets for the specified namespace.
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/namespaces/default/secrets \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+
+HTTP/1.1 200 OK
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+#### API Specification {#secrets-get-specification}
+
+/secrets (GET)  | 
+---------------|------
+description    | Returns the list of secrets for the specified namespace. `id` is the identifying key for the provider to retrieve the secret. `provider` is the name of the provider with the secret.
```

Comment:
Context about `id` and `provider` is probably better suited for the reference doc describing the secret resource.

Reply:
I'll move it there. Thank you.